src/Components/Bill/Bill.jsx

```diff
@@ -2,40 +2,37 @@ import React, { useState } from 'react';
 import './Bill.css';
 
 function Bill({ billData, updateBillData }) {
-  const [showCustomInput, setShowCustomInput] = useState(false);
   const [customTip, setCustomTip] = useState('');
+  const [clickedButton, setClickedButton] = useState("")
 
-  const setValueFromForm = event => {
-    const { name, value } = event.target;
-    let numericalValue;
-
-    if (
-      name === 'tipPercent' &&
-      value !== 'Custom' // Check if the clicked button is not "Custom"
-    ) {
-      setShowCustomInput(false); // Hide the custom input
-      numericalValue = value.includes('%') ? parseFloat(value) : value;
-    } else if (value === 'Custom') {
-      setShowCustomInput(true);
-      numericalValue = '';
+  const setValueFromForm = (eventOrValue) => {
+    let name, value;
+    
+    if (typeof eventOrValue === 'object') {
```

Comment:
I wonder if we could bust this setValueFromFrom into two different functions, that might have been where you were taking us at the end of our coding session? Maybe a separation of responsibility instead of the nested if's??